decode/tinyjpeg.c

```diff
@@ -236,6 +236,8 @@ static int parse_SOF(struct jdec_private *priv, const unsigned char *stream)
     height = be16_to_cpu(stream + 3);
     width  = be16_to_cpu(stream + 5);
     nr_components = stream[7];
+    if(nr_components > COMPONENTS-1)
+        error("Components exceed the value specified by the Jpeg spec\n");
```

Comment:
it should be 1 or 3?

Reply:
> it should be 1 or 3?
The Jpeg Spec indicate that decoders should process scans with 1,2,3 and 4 components, so nr_components should be 0,1,2 and 3.